assets/chat/js/autocomplete.js

```diff
@@ -131,9 +132,13 @@ class ChatAutoComplete {
         this.promoteIfSelected();
         const str = this.input.val().toString();
         const offset = this.input[0].selectionStart + 1;
```

Comment:
I did some troubleshooting myself and suspect the issue lies here.

`pre` should be the contents of the input before the caret, and `post` should be the contents after the caret. For autocompletion at the end of the string, the `+ 1` has no effect. With `substring()`, you just get the entire string when `indexEnd` is greater than the length of the string.

```
console.debug("test string".substring(0, 999));
// test string
```

The `+ 1` has an impact on mid-sentence autocompletion, however. It steals one character from the end of `pre` and adds it to the beginning of `post`, which throws things off. To illustrate with the message in the issue
```
Hello Cake, mid-text dg<caret> auto complete does not work...
```
and when you type the second `g`
```
console.debug(pre);
// `Hello Cake, mid-text dg `
console.debug(char);
// `g`
console.debug(post);
// `auto complete does not work...`
console.debug(pre + char + post);
// `Hello Cake, mid-text dg gauto complete does not work...`
```
Pressing `tab` at this point completes for `g` and resolves to `gachiGASM` rather than `dggL`. It also swallows up the `auto`, but let's set that aside for now (a different bug).

Would you mind testing out yourself to confirm removing the `+ 1` also fixes the issue? If so, I prefer this solution because I think it's simpler and addresses the bug more directly.

Reply:
Yes the simple solution works, good spot I have sorted. :)